### PR TITLE
test/refactor: extract MemoryMetricsCalculator and add unit tests

### PR DESCRIPTION
**Pull request by Manabu-GT**
## Summary by CodeRabbit

* **Refactor**
  * Centralized and improved memory metric calculations (heap, PSS and unit conversion) for more accurate and consistent reporting.
  * Made coroutine dispatching configurable to improve performance and testability.

* **Tests**
  * Added unit and integration tests covering memory calculations and periodic emissions, including timing behavior and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: debugoverlay-core/src/main/kotlin/com/ms/square/debugoverlay/internal/data/source/MemoryDataSource.kt [hunk 1]
```diff
@@ -14,43 +15,37 @@ import kotlinx.coroutines.isActive
...
-    (memoryClassMB * 2).coerceAtLeast(512f) // Minimum 512MB for readability
+    MemoryMetricsCalculator.calculateMaxPss(
+      memoryClass = am.memoryClass,
+      largeMemoryClass = am.largeMemoryClass,
+      isLowRamDevice = am.isLowRamDevice
+    )
   }
 
   fun heapUsage(interval: Duration = 1L.seconds): Flow<Percentage> = flow {
     while (currentCoroutineContext().isActive) {
```
[the remark this answers is not in the document]
Comment (Manabu-GT) [reply debugoverlay-core/src/main/kotlin/com/ms/square/debugoverlay/internal/data/source/MemoryDataSource.kt]: While technically correct that delay() checks cancellation, the explicit isActive check:                                                                                                                                                    
  - Documents intent clearly                                                                                                                                                                                                                  
  - Matches the pattern used elsewhere in the codebase (FpsDataSource)                                                                                                                                                                        
  - Has negligible overhead                                                                                                                                                                                                                   
  - Safer if someone adds code before delay() later

File: debugoverlay-core/src/test/kotlin/com/ms/square/debugoverlay/internal/data/source/MemoryMetricsCalculatorTest.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      maxMemory = 0L
+    )
+    val negativeResult = MemoryMetricsCalculator.calculateHeapUsagePercentage(
+      usedMemory = 50_000_000L,
+      maxMemory = -1L
+    )
+
+    assertThat(zeroResult).isEqualTo(Percentage.ZERO)
+    assertThat(negativeResult).isEqualTo(Percentage.ZERO)
+  }
```
[the remark this answers is not in the document]
Comment (Manabu-GT) [reply debugoverlay-core/src/test/kotlin/com/ms/square/debugoverlay/internal/data/source/MemoryMetricsCalculatorTest.kt]: We already discussed this - clamping is Percentage.ofClamped()'s responsibility. The calculator just passes the ratio through. Testing clamping here tests the wrong layer.
